staging/src/k8s.io/sample-controller/artifacts/examples/crd.yaml

```diff
@@ -1,10 +1,33 @@
-apiVersion: apiextensions.k8s.io/v1beta1
+apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   name: foos.samplecontroller.k8s.io
+  # for more information on the below annotation, please see
+  # https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/2337-k8s.io-group-protection/README.md
+  annotations:
+    "api-approved.kubernetes.io": "unapproved, experimental-only; please get an approval from Kubernetes API reviewers if you're trying to develop a CRD in the *.k8s.io or *.kubernetes.io groups"
 spec:
   group: samplecontroller.k8s.io
-  version: v1alpha1
+  versions:
+    - name: v1alpha1
+      served: true
+      storage: true
+      schema:
+        # schema used for validation
+        openAPIV3Schema:
+          type: object
+          properties:
+            spec:
+              type: object
+              properties:
+                deploymentName:
+                  type: string
+                image:
+                  type: string
+                replicas:
+                  type: integer
+                  minimum: 1
+                  maximum: 10
```

Comment:
I forgot if [pruning](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#field-pruning) works on `status`. Let me verify

Reply:
yes, pruning works on `status`:

having a CR with status in etcd:
```
$ kubectl get foo my-foo -oyaml
apiVersion: samplecontroller.k8s.io/v1alpha1
kind: Foo
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"samplecontroller.k8s.io/v1alpha1","kind":"Foo","metadata":{"annotations":{},"name":"my-foo","namespace":"default"},"spec":{"deploymentName":"my-deployment","replicas":1},"status":{"availableReplicas":1}}
  creationTimestamp: "2021-05-12T18:07:50Z"
  generation: 1
  name: my-foo
  namespace: default
  resourceVersion: "537"
  uid: 920b53d2-baee-473d-9d25-bcf8dbdb8294
spec:
  deploymentName: my-deployment
  replicas: 1
status:
  availableReplicas: 1
```
edit the CRD to omit status in the schema
```
$ cat crd.yaml
apiVersion: apiextensions.k8s.io/v1
kind: CustomResourceDefinition
metadata:
  name: foos.samplecontroller.k8s.io
  # for more information on the below annotation, please see
  # https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/2337-k8s.io-group-protection/README.md
  annotations:
    "api-approved.kubernetes.io": "unapproved, experimental-only; please get an approval from Kubernetes API reviewers if you're trying to develop a CRD in the *.k8s.io or *.kubernetes.io groups"
spec:
  group: samplecontroller.k8s.io
  versions:
    - name: v1alpha1
      served: true
      storage: true
      schema:
        # schema used for validation
        openAPIV3Schema:
          type: object
          properties:
            spec:
              type: object
              properties:
                deploymentName:
                  type: string
                image:
                  type: string
                replicas:
                  type: integer
                  minimum: 1
                  maximum: 10
      # subresources for the custom resource
      subresources:
        # enables the status subresource
        status: {}
  names:
    kind: Foo
    plural: foos
  scope: Namespaced
```
read the CR again. Note the `status` is pruned
```
$ kubectl get foo my-foo -oyaml
apiVersion: samplecontroller.k8s.io/v1alpha1
kind: Foo
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"samplecontroller.k8s.io/v1alpha1","kind":"Foo","metadata":{"annotations":{},"name":"my-foo","namespace":"default"},"spec":{"deploymentName":"my-deployment","replicas":1},"status":{"availableReplicas":1}}
  creationTimestamp: "2021-05-12T18:07:50Z"
  generation: 1
  name: my-foo
  namespace: default
  resourceVersion: "537"
  uid: 920b53d2-baee-473d-9d25-bcf8dbdb8294
spec:
  deploymentName: my-deployment
  replicas: 1
```

Reply:
please add the following to the schema in both `crd.yaml` and `crd-status-subresource.yaml`
```
            status:
              type: object
              properties:
                availableReplicas:
                  type: integer
```